baseplate/context/redis.py

```diff
@@ -148,3 +150,66 @@ def __init__(self, trace_name, server_span, connection_pool,
     def execute(self, **kwargs):
         with self.server_span.make_child(self.trace_name):
             return super(MonitoredRedisPipeline, self).execute(**kwargs)
+
+
+class MessageQueue(object):
+    """A redis-backed variant of :py:class:`~baseplate.message_queue.MessageQueue`.
+
+    :param ``name`` can be any string.
+
+    :param ``client`` should be a :py:class:`redis.ConnectionPool` or
+    :py:class:`redis.BlockingConnectionPool` from which a client connection can be
+    created from (preferrably generated from the :py:func:`pool_from_config` helper).
+
+    """
+    def __init__(self, name, client):
+        self.queue = name
+        if isinstance(client, redis.BlockingConnectionPool) or \
+                isinstance(client, redis.ConnectionPool):
+            self.client = redis.Redis(connection_pool=client)
+        else:
+            self.client = client
+
+    def get(self, timeout=None):
+        """Read a message from the queue.
+
+        :param int timeout: If the queue is empty, the call will block up to
+            ``timeout`` seconds or forever if ``None``, if a float is given,
+            it will be rounded up to be an integer
+        :raises: :py:exc:`TimedOutError` The queue was empty for the allowed
+            duration of the call.
+
+        """
+        if isinstance(timeout, float):
+            timeout = int(ceil(timeout))
+
+        if timeout == 0:
+            message = self.client.lpop(self.queue)
+        else:
+            message = self.client.blpop(self.queue, timeout=timeout)
+
+            if message:  # BLPOP returns a tuple of (key, value) and we just want value
+                message = message[1]
+
+        if not message:
+            raise message_queue.TimedOutError
+
+        return pickle.loads(message)
```

Comment:
I'd really prefer to avoid pickle if we can. It's magical, hard to use with other languages involved, and has security complications. Maybe JSON if we really need to serialize? parameterize the serializer on this class? or just make it explicit that the client of this class is to handle serialization? (I think the posix one is all strings too right?)